src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -263,14 +264,24 @@ object DepositHandler {
       }
     }
 
+    val itemsToResolve = fetchItems diff fetchItemsInBagStore
     for {
-      _ <- resolveFetchItems(bagitDir, fetchItems diff fetchItemsInBagStore)
+      _ <- resolveFetchItems(bagitDir, itemsToResolve)
+      _ <- removeFromFetchTxt(bagitDir, itemsToResolve)
       bag = getBagFromDir(bagitDir)
       validationResult = bag.verifyValid
       _ <- handleValidationResult(bag, validationResult, fetchItemsInBagStore)
     } yield ()
   }
 
+  private def removeFromFetchTxt(bagDir: File, items: Seq[FetchTxt.FilenameSizeUrl]): Try[Unit] = Try {
+    getFetchTxt(bagDir).foreach { case fetchTxt =>
```

Comment:
Seems more obvious to me. In general I'm always careful with the use of `foreach`, as it does not return anything. In the `items.foreach(fetchTxt.remove)` it is a logical choice to do so, but to me it is not at the `getFetchTxt.foreach`. I'd say you want a `Try` back, so you use `map`.

Also, the `getOrElse` makes it more explicitly clear that it is still OK if the `getFetchTxt` returns a `None`. For that to work you also require a `map`.

Sidenote: you could also merge the two `map`s and just return a `()` at the end of the first `map`. The reason I choose for the approach with the second `map` is purely esthetic.

Since each of the three parts of the first `map` are functions that might throw an exception (welcome to the stupidly impure world of Java!), we require this `Try` around it to catch all exceptions and return in `Failure` rather than a `Success` if an exception is thrown.